lib/src/connection.dart

```diff
@@ -165,9 +165,7 @@ abstract class Connection {
 
     // Setup frame writing.
     _frameWriter = FrameWriter(_hpackContext.encoder, outgoing, peerSettings);
-    _frameWriter.doneFuture.then((_) {
-      _terminate(ErrorCode.CONNECT_ERROR, causedByTransportError: true);
-    }).catchError((error, stack) {
+    _frameWriter.doneFuture.whenComplete(() {
       _terminate(ErrorCode.CONNECT_ERROR, causedByTransportError: true);
```

Comment:
The doc comment mentions that the Future returned by `_terminate` will never complete with an error.

https://github.com/dart-lang/http2/blob/9523cfac735f54f280fbeb310af75a5535d895d6/lib/src/connection.dart#L402